crates/pvm_bindings/src/callbacks.rs

```diff
@@ -0,0 +1,77 @@
+use std::ffi::c_void;
+
+/// Callback type for handling external calls with 3 parameters
+#[repr(C)]
+pub struct ExternalCallCallback3 {
```

Comment:
Maybe have a macro to generate those? Not sure Rust's type system is expressive enough to use generics here.